src/tracing/agent.cc

```diff
@@ -0,0 +1,68 @@
+#include "tracing/agent.h"
+
+#include "env-inl.h"
+#include "libplatform/libplatform.h"
+#include "tracing/trace_config_parser.h"
+
+namespace node {
+namespace tracing {
+
+Agent::Agent(Environment* env) : parent_env_(env) {}
+
+void Agent::Start(v8::Platform* platform, const char* trace_config_file) {
+  auto env = parent_env_;
+  platform_ = platform;
+
+  int err = uv_loop_init(&tracing_loop_);
+  CHECK_EQ(err, 0);
+
+  NodeTraceWriter* trace_writer = new NodeTraceWriter(&tracing_loop_);
+  TraceBuffer* trace_buffer = new NodeTraceBuffer(
+      NodeTraceBuffer::kBufferChunks, trace_writer, &tracing_loop_);
+
+  tracing_controller_ = new TracingController();
+
+  TraceConfig* trace_config = new TraceConfig();
+  if (trace_config_file) {
+    std::ifstream fin(trace_config_file);
+    std::string str((std::istreambuf_iterator<char>(fin)),
+                    std::istreambuf_iterator<char>());
+    TraceConfigParser::FillTraceConfig(env->isolate(), trace_config,
+                                       str.c_str());
```

Comment:
It looks like the above code will crash if the trace config file is not found or has invalid contents. Is that acceptable? Or should there be a more helpful error message?